hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/ConfigurationProperties.java

```diff
@@ -94,6 +109,23 @@ public Map<String, String> getPropertiesWithPrefix(
     return properties;
   }
 
+  /**
+   * Update or create value in the nodes
+   * @param name name of the property
+   * @param value value of the property
+   */
+  public void set(String name, String value) {
+    getNode(name).ifPresent(node -> node.getValues().put(name, value));
+  }
+
+  /**
+   * Delete value from nodes
+   * @param name name of the property
+   */
+  public void unset(String name) {
+    getNode(name).ifPresent(node -> node.getValues().remove(name));
```

Comment:
What happens when a node does not have any more children or values left? We need some sort of garbage collection here to avoid memory leaks.